### PR TITLE
add hooks remove hocs 

### PR DESCRIPTION
**Pull request by omarb147**

### CODE REVIEW

File: src/module/node/reducer.ts [hunk 1]
```diff
@@ -1,36 +1,26 @@
 import { getType } from 'typesafe-actions';
 import { INodesState } from '@/type/types';
-import { resizeNode, nodeActionTypes } from './actions';
+import uniqid from 'uniqid';
```
Comment (alexwong-crc) [inline src/module/node/reducer.ts]: never used uniqid before, how does it differ from uuid?

File: src/components/Node/node.tsx [hunk 2]
```diff
@@ -1,42 +1,23 @@
 import React from 'react';
-import { compose } from 'recompose';
-import { Rnd, RndResizeCallback } from 'react-rnd';
-import { withNodes, WithNodesProps } from '@/hoc/withNodes';
+import { Rnd } from 'react-rnd';
```
Comment (alexwong-crc) [inline src/components/Node/node.tsx]: what is react-rnd?

File: src/hoc/useNodes.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { useSelector as reduxUseSelector, useDispatch, TypedUseSelectorHook } from 'react-redux';
+import { addNode, AddNode } from '@/module/node/actions';
+import { IGraphState, INodesState } from '@/type/types';
+
+const useSelector: TypedUseSelectorHook<IGraphState> = reduxUseSelector;
```
Comment (alexwong-crc) [inline src/hoc/useNodes.tsx]: I don't understand what is happening here? 

useSelector is imported under a different alias, then we declare a variable with the same name that is the same as the imported useSelector, and then we pass a select function to it. 

Can't we just use the useSelector out of the box?

File: src/module/node/actions.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
-  height: string;
-  width: string;
+export const addNode = createAction('nodes/ADD_NODE')<{
+  text: string;
+  color: string;
 }>();
+export type AddNode = ActionType<typeof addNode>;
 
-export type nodeActionTypes = ActionType<typeof resizeNode>;
+export type nodeActionTypes = AddNode;
```
Comment (alexwong-crc) [inline src/module/node/actions.ts]: isn't line 7 and 9 the same?

**Comment by alexwong-crc** [comment]
When I pulled it to my local, there was an error with the hoc/withNodes file. Is it still needed?

File: src/hoc/useNodes.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+export default () => {
+  // Replaces mapStateToProps
+  const nodes: INodesState = useSelector((state) => state.nodes);
```
Comment (alexwong-crc) [inline src/hoc/useNodes.tsx]: Just pulled the branch to play with it. 
Do we even need hoc's anymore? Can we just export the individual hooks and import it to the file we need them in?

File: src/module/node/reducer.ts (hunk 1 above)
Comment (omarb147) [reply src/module/node/reducer.ts]: I don't think there is much difference just generate unique ID - can use uuid if you prefer longer term ?

**Pull request merged**